### PR TITLE
Remove IP verifications in SetPools

### PR DESCRIPTION
**Pull request by fedepaol**
This is https://github.com/metallb/metallb/pull/1948 plus the fixes suggested in the last phase of the PR.

Pool changes might invalidate the LB IPs of services. When this happens, the pool controller ignores the new configuration and continues operating with the current one. This PR modifies this behavior by always accepting the new configuration and letting the service controller to clear the IPs of the affected services.

Fixes https://github.com/metallb/metallb/issues/462

### CODE REVIEW

**Comment by fedepaol** [comment]
cc @oribon

File: controller/controller_test.go [hunk 1]
```diff
@@ -1121,6 +1132,102 @@ func TestDeleteRecyclesIP(t *testing.T) {
 		t.Fatal("svc2 didn't get an IP")
 	}
 }
+func TestControllerReassing(t *testing.T) {
```
Comment (oribon) [inline controller/controller_test.go:1135]: nit: reassign?

**Review by oribon — COMMENTED** [review]
small nit, lgtm

File: controller/controller_test.go (hunk 1 above)
Comment (fedepaol) [reply controller/controller_test.go]: done

**Pull request merged**